go/store/types/edits/disk_backed.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package edits
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/google/uuid"
+
+	"github.com/dolthub/dolt/go/store/atomicerr"
+	"github.com/dolthub/dolt/go/store/types"
+)
+
+var _ types.EditAccumulator = (*DiskBackedEditAcc)(nil)
+
+// DiskBackedEditAcc is an EditAccumulator implementation that flushes the edits to disk at regular intervals
+type DiskBackedEditAcc struct {
+	ctx        context.Context
+	nbf        *types.NomsBinFormat
+	vrw        types.ValueReadWriter
+	directory  string
+	newEditAcc func() types.EditAccumulator
+
+	backing types.EditAccumulator
+
+	accumulated   int64
+	flushInterval int64
+
+	ae *atomicerr.AtomicError
+	wg *sync.WaitGroup
+
+	flushResults chan string
```

Comment:
Feel like flushResults is a confusing name. Perhaps flushedFiles since we're returning file paths